src/map/script.c

```diff
@@ -1903,8 +1943,15 @@ static const char *parse_syntax(const char *p)
 		} else if( p2 - p == 8 && strncmp(p, "function", 8) == 0 ) {
 			// internal script function
 			const char *func_name;
+			bool is_extern = false;
 
 			func_name = script->skip_space(p2);
+
+			if (*func_name == ':') {
```

Comment:
to mark functions as exported I chose `:` because it is short and easy to remember, but we could use a keyword if this is preferred, such as:
* `extern function MyFunction { … }`
* `export function MyFunction { … }`
* `public function MyFunction { … }`